client/src/context/productState.js

```diff
@@ -26,12 +27,32 @@ const ProductState = props => {
         } catch (err) {}
     };
 
+    const loadAllProducts = async filterObj => {
+        if (localStorage.getItem("token")) {
+            axios.defaults.headers.common[
+                "x-auth-token"
+            ] = localStorage.getItem("token");
+        } else delete axios.defaults.headers.common["x-auth-token"];
+        try {
+            const query = `http://localhost:3001/product/pagination/products`;
```

Comment:
this should remain as 

```suggestion
            const request = `http://localhost:3001/products`;
```

for all products .

The `filterObj` is your query